pkg/json_raw_results.go

```diff
@@ -46,6 +46,24 @@ type jsonTool struct {
 	// TODO: Runs, Issues, Merge requests.
 }
 
+type jsonBranchProtectionSettings struct {
+	RequiredApprovingReviewCount        *int     `json:"required-reviewer-count"`
+	AllowsDeletions                     *bool    `json:"allows-deletions"`
+	AllowsForcePushes                   *bool    `json:"allows-force-pushes"`
+	RequiresCodeOwnerReviews            *bool    `json:"requires-code-owner-review"`
+	RequiresLinearHistory               *bool    `json:"required-linear-history"`
+	DismissesStaleReviews               *bool    `json:"dismisses-stale-reviews"`
+	EnforcesAdmins                      *bool    `json:"enforces-admin"`
+	RequiresStatusChecks                *bool    `json:"requires-status-checks"`
+	RequiresUpToDateBranchBeforeMerging *bool    `json:"requires-updated-branches-to-merge"`
+	StatusCheckContexts                 []string `json:"status-checks-contexts"`
+}
+
+type jsonBranchProtection struct {
+	Protection *jsonBranchProtectionSettings `json:"protection"`
```

Comment:
we use a pointer: if the branch is not protected, nothing is populated and `Protection` is set to `nil`